Update HTMLFastPathParser::scanTagName() to return an ElementName

cdumez opened the pull request:
#### 8e6de7844cbbe195f4f71cec376dba02c6ad2fd2
<pre>
Update HTMLFastPathParser::scanTagName() to return an ElementName
<a href="https://bugs.webkit.org/show_bug.cgi?id=254857">https://bugs.webkit.org/show_bug.cgi?id=254857</a>

Reviewed by Darin Adler.

Update HTMLFastPathParser::scanTagName() to return an ElementName enum instead
of a Span&lt;Character&gt;. This simplifies the logic in the HTML fast parser and is
slightly more efficient.

* Source/WebCore/dom/make_names.pl:
(printElementNameHeaderFile):
(printElementNameCppFile):
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::HTMLFastPathParser::scanTagName):
(WebCore::HTMLFastPathParser::parseSpecificElements):
(WebCore::HTMLFastPathParser::parseElement):
(WebCore::onlyContainsLowercaseASCIILetters): Deleted.
(WebCore::tagNameHash): Deleted.

Canonical link: <a href="https://commits.webkit.org/262484@main">https://commits.webkit.org/262484@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1f6983c7cbeb67ee11cc6ae4bbf10ae9d142837c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1618 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1649 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1706 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2539 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1738 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1597 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1714 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1712 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1520 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1632 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1469 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1466 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2376 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1466 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1452 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1366 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1476 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/1494 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2541 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1512 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1353 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1428 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1459 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/408 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1575 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/dom/make_names.pl:
```diff
@@ -1048,6 +1048,8 @@ sub printElementNameHeaderFile
     print F "\n";
     print F "ElementName findElementName(Namespace, const String&);\n";
+    print F "ElementName findHTMLElementNameFromBuffer(Span<const LChar>);\n";
+    print F "ElementName findHTMLElementNameFromBuffer(Span<const UChar>);\n";
```
Can these just be `findHTMLElementName`?  I don’t think we need “from buffer” in the name.

darinadler commented on Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
```diff
@@ -845,16 +815,16 @@ class HTMLFastPathParser {
     template<class... Tags> RefPtr<Element> parseSpecificElements()
     {
-        CharSpan tagName = scanTagName();
+        ElementName tagName = scanTagName();
```
`auto`

cdumez committed:
Update HTMLFastPathParser::scanTagName() to return an ElementName
https://bugs.webkit.org/show_bug.cgi?id=254857

Reviewed by Darin Adler.

Update HTMLFastPathParser::scanTagName() to return an ElementName enum instead
of a Span<Character>. This simplifies the logic in the HTML fast parser and is
slightly more efficient.

* Source/WebCore/dom/make_names.pl:
(printElementNameHeaderFile):
(printElementNameCppFile):
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::HTMLFastPathParser::scanTagName):
(WebCore::HTMLFastPathParser::parseSpecificElements):
(WebCore::HTMLFastPathParser::parseElement):
(WebCore::onlyContainsLowercaseASCIILetters): Deleted.
(WebCore::tagNameHash): Deleted.

Canonical link: https://commits.webkit.org/262484@main
